h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java

```diff
@@ -22,7 +22,37 @@ protected void readModelData() throws IOException {
     _model._ntrees_per_group = tpc;
     _model._compressed_trees = new byte[_model._ntree_groups * tpc][];
     _model._mojo_version = ((Number) readkv("mojo_version")).doubleValue();
-
+    if (_model._mojo_version < 1.40) {
+        _model._genmodel_encoding = "AUTO";
+    } else {
+        _model._genmodel_encoding = readkv("_genmodel_encoding").toString();
+        _model._orig_projection_array = readkv("_orig_projection_array");
+        Integer n = readkv("_n_orig_names");
+        if (n != null) {
```

Comment:
this is always true, maybe test > 0 ?

Reply:
it's not, it is needed like this. For example EnumEncoder doesn't need originalNames, so doesn't have them set, so it writes this as null